src/Pages/Main/MainBanner/MainBanner.component.js

```diff
@@ -0,0 +1,170 @@
+import React, { Component } from "react";
+import { Link } from "react-router-dom";
+
+import "./MainBanner.styles.scss";
+
+class MainBanner extends Component {
+  constructor() {
+    super();
+    this.state = {
+      bannerImages: [],
+      bannerImgWidth: window.innerWidth,
+      bannerImgCount: 0,
+      currentBannerId: 1,
+      bannerImgsXcoordinate: 0,
+      bannerImgsTransitionDelay: 800,
+      leftClickedOnMainBanner: false,
+      slideIntervalDelay: 4000,
+    };
+  }
+
+  componentDidMount = () => {
+    const { slideIntervalDelay } = this.state;
+    fetch("http://localhost:3000/data/main/MainBannerImagesData.json")
+      .then((res) => res.json())
+      .then((res) => {
+        this.setState({
+          bannerImages: res.bannerImages,
+          bannerImgCount: res.bannerImages.length,
+        });
+      });
+    this.timerId = setInterval(this.bannerSlideAuto, slideIntervalDelay);
+  };
+
+  componentDidUpdate = (prevProps, prevState) => {
+    const { currentBannerId, slideIntervalDelay } = this.state;
+    if (currentBannerId !== prevState.currentBannerId) {
+      clearInterval(this.timerId);
+      this.timerId = setInterval(this.bannerSlideAuto, slideIntervalDelay);
+    }
+  };
+
+  componentWillUnmount = () => {
+    clearInterval(this.timerId);
+  };
+
+  bannerSlideAuto = () => {
+    const {
+      bannerImages,
+      bannerImgWidth,
+      bannerImgsXcoordinate,
+      currentBannerId,
+      leftClickedOnMainBanner,
+    } = this.state;
+    const xIncrement = bannerImgWidth * -1;
+    this.setState({
+      currentBannerId:
+        leftClickedOnMainBanner && currentBannerId === bannerImages.length
+          ? currentBannerId - 1
+          : !leftClickedOnMainBanner && currentBannerId === bannerImages.length
+          ? 1
+          : currentBannerId + 1,
+      bannerImgsXcoordinate:
+        leftClickedOnMainBanner && currentBannerId === bannerImages.length
+          ? bannerImgsXcoordinate - xIncrement
+          : !leftClickedOnMainBanner && currentBannerId === bannerImages.length
+          ? 0
+          : bannerImgsXcoordinate + xIncrement,
+      bannerImgsTransitionDelay:
+        !leftClickedOnMainBanner && currentBannerId === bannerImages.length
+          ? 0
+          : 800,
+      slideIntervalDelay:
+        !leftClickedOnMainBanner && currentBannerId === bannerImages.length
+          ? 0
+          : 4000,
+      leftClickedOnMainBanner: false,
+    });
```

Comment:
자주 쓰는 조건을 변수로 빼서 관리하면 더 가독성이 좋아질 거 같아요!
변수 이름만 적절하게 바꿔서 사용해주세요! ㅋㅋ
const isValid = leftClickedOnMainBanner && currentBannerId === bannerImages.length;
const isValid2 = !leftClickedOnMainBanner && currentBannerId === bannerImages.length;
```suggestion
    this.setState({
      currentBannerId: isValid ? currentBannerId - 1
          : isValid2
          ? 1
          : currentBannerId + 1,
      bannerImgsXcoordinate:
        leftClickedOnMainBanner && currentBannerId === bannerImages.length
          ? bannerImgsXcoordinate - xIncrement
          : !leftClickedOnMainBanner && currentBannerId === bannerImages.length
          ? 0
          : bannerImgsXcoordinate + xIncrement,
      bannerImgsTransitionDelay:
        !leftClickedOnMainBanner && currentBannerId === bannerImages.length
          ? 0
          : 800,
      slideIntervalDelay:
        !leftClickedOnMainBanner && currentBannerId === bannerImages.length
          ? 0
          : 4000,
      leftClickedOnMainBanner: false,
    });
```

Reply:
오우 적용했습니다~ 훨씬 깔끔해졌어요 😄